src/PooledArrays.jl

```diff
@@ -288,10 +350,28 @@ Base.sort(pa::PooledArray; kw...) = pa[sortperm(pa; kw...)]
 ##
 ##############################################################################
 
-Base.convert(::Type{PooledArray{S,R1,N}}, pa::PooledArray{T,R2,N}) where {S,T,R1<:Integer,R2<:Integer,N} =
-    PooledArray(RefArray(convert(Array{R1,N}, pa.refs)), convert(Dict{S,R1}, pa.invpool))
-Base.convert(::Type{PooledArray{S,R,N}}, pa::PooledArray{T,R,N}) where {S,T,R<:Integer,N} =
-    PooledArray(RefArray(copy(pa.refs)), convert(Dict{S,R}, pa.invpool))
+function Base.convert(::Type{PooledArray{S,R1,N}}, pa::PooledArray{T,R2,N}) where {S,T,R1<:Integer,R2<:Integer,N}
+    if S === R && R1 === R2
+        return pa
+    else
+        refs_conv = convert(Array{R1,N}, pa.refs)
+        @assert refs_conv !== pa.refs
+        invpool_conv = convert(Dict{S,R}, pa.invpool)
+        @assert invpool_conv !== pa.invpool
+        return PooledArray(RefArray(refs_conv), invpool_conv)
+    end
+end
+
+function Base.convert(::Type{PooledArray{S,R,N}}, pa::PooledArray{T,R,N}) where {S,T,R<:Integer,N}
```

Comment:
Do we actually need a separate function from the one above? (These are really tricky, in CategoricalArrays it's been a nightmare to get right.)

Reply:
I will try removing it. It might lead to dispatch ambiguities, but hopefully it will not.